src/plugins/charts/public/plugin.ts

```diff
@@ -42,8 +42,8 @@ export class ChartsPlugin implements Plugin<ChartsPluginSetup, ChartsPluginStart
     dependencies.expressions.registerFunction(palette);
     dependencies.expressions.registerFunction(systemPalette);
     this.themeService.init(core.theme);
-    this.palettes = this.paletteService.setup();
-
+    // TODO: make this reactive to changes in $theme
```

Comment:
Please create a follow up issue to track this

Reply:
https://github.com/elastic/kibana/issues/203710